backend/src/services/auth/auth-password-service.ts

```diff
@@ -114,26 +119,35 @@ export const authPaswordServiceFactory = ({
    * Email password reset flow via email. Step 1 send email
    */
   const sendPasswordResetEmail = async (email: string) => {
+    const startTime = Date.now();
+
     const user = await userDAL.findUserByUsername(email);
-    // ignore as user is not found to avoid an outside entity to identify infisical registered accounts
-    if (!user || (user && !user.isAccepted)) return;
 
-    const cfg = getConfig();
-    const token = await tokenService.createTokenForUser({
-      type: TokenType.TOKEN_EMAIL_PASSWORD_RESET,
-      userId: user.id
-    });
+    if (user && user.isAccepted) {
+      const cfg = getConfig();
+      const token = await tokenService.createTokenForUser({
+        type: TokenType.TOKEN_EMAIL_PASSWORD_RESET,
+        userId: user.id
+      });
 
-    await smtpService.sendMail({
-      template: SmtpTemplates.ResetPassword,
-      recipients: [email],
-      subjectLine: "Infisical password reset",
-      substitutions: {
-        email,
-        token,
-        callback_url: cfg.SITE_URL ? `${cfg.SITE_URL}/password-reset` : ""
-      }
-    });
+      await smtpService.sendMail({
+        template: SmtpTemplates.ResetPassword,
+        recipients: [email],
+        subjectLine: "Infisical password reset",
+        substitutions: {
+          email,
+          token,
+          callback_url: cfg.SITE_URL ? `${cfg.SITE_URL}/password-reset` : ""
+        }
+      });
+    }
+
+    const elapsedTime = Date.now() - startTime;
```

Comment:
What you can do here is:

Wrap the thing in async and instead of await use then and catch. This makes the process a background task.

Return straight away a message stating a reset confirmation mail has been send. This is how some of the existing platform like mongodb, sendgrid

Reply:
Waiting 8s is also kinda weird and cause user frustration